weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/callsTableColumns.tsx

```diff
@@ -299,135 +302,19 @@ function buildCallsTableColumns(
-    }
-    return node;
-  }) as GridColumnGroupingModel;
-
-  for (const key of filteredDynamicColumnNames) {
-    const col: GridColDef<TraceCallSchema> = {
-      flex: 1,
-      minWidth: 150,
-      field: key,
-      // CPR (Tim) - (BackendExpansion): This can be removed once we support backend expansion!
-      filterable: !columnIsRefExpanded(key) && !columnsWithRefs.has(key),
-      // CPR (Tim) - (BackendExpansion): This can be removed once we support backend expansion!
-      sortable: !columnIsRefExpanded(key) && !columnsWithRefs.has(key),
-      filterOperators: allOperators,
-      headerName: key,
-      renderHeader: () => {
-        return (
-          <div
-            style={{
-              fontWeight: 600,
-            }}>
-            {key.split('.').slice(-1)[0]}
-          </div>
-        );
-      },
-      valueGetter: cellParams => {
-        const val = (cellParams.row as any)[key];
-        if (Array.isArray(val) || typeof val === 'object') {
-          try {
-            return JSON.stringify(val);
-          } catch {
-            return val;
-          }
-        }
-        return val;
-      },
-      renderCell: cellParams => {
-        const val = (cellParams.row as any)[key];
-        if (val === undefined) {
-          return <NotApplicable />;
-        }
-        return (
-          <ErrorBoundary>
-            {/* In the future, we may want to move this isExpandedRefWithValueAsTableRef condition
-            into `CallValue`. However, at the moment, `ExpandedRefWithValueAsTableRef` is a
-            CallsTable-specific data structure and we might not want to leak that into the
-            rest of the system*/}
-            {isExpandedRefWithValueAsTableRef(val) ? (
-              <SmallRef objRef={parseRef(val[EXPANDED_REF_REF_KEY])} />
-            ) : (
-              <CellValue value={val} />
-            )}
-          </ErrorBoundary>
-        );
-      },
-    };
-
-    if (groupIds.has(key)) {
-      col.renderHeader = () => {
-        return <></>;
-      };
-    } else if (expandedRefCols.has(key)) {
-      col.renderHeader = () => {
-        return (
-          <CollapseHeader
-            headerName={key.split('.').slice(-1)[0]}
-            field={key}
-            onCollapse={onCollapse}
-          />
-        );
-      };
-    } else if (columnsWithRefs.has(key)) {
-      col.renderHeader = () => {
-        return (
-          <ExpandHeader
-            headerName={key.split('.').slice(-1)[0]}
-            field={key}
-            hasExpand
-            onExpand={onExpand}
-          />
-        );
-      };
-    }
-    cols.push(col);
-  }
+  const {cols: newCols, groupingModel} = buildDynamicColumns<TraceCallSchema>(
+    filteredDynamicColumnNames,
+    (row, key) => (row as any)[key],
+    key => expandedRefCols.has(key),
+    key => columnsWithRefs.has(key),
+    onCollapse,
+    onExpand,
+    // CPR (Tim) - (BackendExpansion): This can be removed once we support backend expansion!
```

Comment:
What does `CPR` stand for?

Reply:
TODO from "Calls Page Refactor"... changing to todo